### PR TITLE
Marker Interface Usage

### PR DESCRIPTION
**Pull request by dmdashenkov**
Since recently, all the special-case message types (such as events, commands, and rejections) marked with special interfaces (`EventMessage`, `CommandMessage`, and `RejectionMessage` respectively).

In this PR we adjust the framework API in order to use those interfaces.

The goal is to make the framework API (including the internal APIs) less prone to misunderstanding and typo-driven errors.

Also, these are some API changed made due to the marked interfaces integration:
 - returning `Empty` from handler methods is now completely forbidden; instead, use `NothingHappened` as a stub event;
 - importing events into an `Aggregate` history by returning an `io.spine.core.Event` from a handler/reactor method is now impossible.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

The tests are to be fixed soon.

File: server/src/test/java/io/spine/server/event/error/package-info.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+@CheckReturnValue
+@ParametersAreNonnullByDefault
+package io.spine.server.event.error;
```
Comment (armiol) [inline server/src/test/java/io/spine/server/event/error/package-info.java]: Please have a few words on the package.

File: server/src/test/java/io/spine/server/model/noops/NothingHappenedTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.server.model.noops.given.NoOpMessageTestEnv.archiveSingleFile;
+import static io.spine.testing.client.blackbox.Count.count;
+import static io.spine.testing.server.blackbox.VerifyCommands.emittedCommand;
+import static io.spine.testing.server.blackbox.VerifyEvents.emittedEvent;
+
+@DisplayName("When NothingHappened event is emitted")
+class NothingHappenedTest {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/model/noops/NothingHappenedTest.java]: While we don't remove the `@author` tags at all, let's have it and document the class purpose.

I'd document some of the test classes and package info files in this changeset as well.

File: server/src/test/java/io/spine/server/projection/ProjectionShould.java [hunk 3]
```diff
@@ -108,31 +111,31 @@ void exposeEventClasses() {
...
         TestEventFactory eventFactory = TestEventFactory.newInstance(getClass());
-        StringValue strValue = StringValue.newBuilder()
-                                          .setValue("eins zwei drei")
-                                          .build();
-        Int32Value intValue = Int32Value.newBuilder()
-                                        .setValue(123)
-                                        .build();
+        StringImported stringImported = StringImported
+                .newBuilder()
+                .setValue("eins zwei drei")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/projection/ProjectionShould.java]: I thinks `ein` is more common.

File: server/src/test/proto/spine/test/aggregate/archiver/commands.proto [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+package spine.test.aggregate.archiver;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.test.aggregate.archiver";
+option java_outer_classname = "CommandsProto";
+option java_multiple_files = true;
+
+message Evaluate {
```
Comment (armiol) [inline server/src/test/proto/spine/test/aggregate/archiver/commands.proto]: How come this is an "archiver"? I'd use something from zip/unzip terms.

File: server/src/test/proto/spine/test/commandservice/unknown_commands.proto [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+package spine.test.commandservice;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.test.commandservice";
+option java_outer_classname = "UnknownCommandsProto";
+option java_multiple_files = true;
+
+message CmdSDontHandle {
```
Comment (armiol) [inline server/src/test/proto/spine/test/commandservice/unknown_commands.proto]: I think `S` is redundant.

File: testutil-server/src/main/java/io/spine/testing/server/blackbox/EmittedMessages.java [hunk 6]
```diff
@@ -37,13 +37,15 @@
  * @param <W> the type of the wrapper object containing messages
  * @author Alexander Yevsyukov
  */
-public abstract class EmittedMessages<C extends MessageClass, W extends Message> {
+public abstract class EmittedMessages<C extends MessageClass<M>,
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/testing/server/blackbox/EmittedMessages.java]: Let's document the newly added generic param.

File: testutil-server/src/main/java/io/spine/testing/server/blackbox/MessageTypeCounter.java [hunk 7]
```diff
@@ -40,13 +40,14 @@
  * @author Mykhailo Drachuk
  * @author Alexander Yevsyukov
  */
-class MessageTypeCounter<C extends MessageClass, W extends Message> {
+class MessageTypeCounter<C extends MessageClass, W extends Message, M extends Message> {
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/testing/server/blackbox/MessageTypeCounter.java]: Let's document the newly added generic param.

File: testutil-server/src/test/proto/spine/testing/server/log/events.proto [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+package spine.testing.server.log;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.testing.server.log";
+option java_outer_classname = "EventsProto";
+option java_multiple_files = true;
+
+message FloatLogged {
```
Comment (armiol) [inline testutil-server/src/test/proto/spine/testing/server/log/events.proto]: It's weird to have `LogInteger` command with no related event and `FloatLogged` event with no origin command.

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Pull request merged**